src/librustdoc/lint.rs

```diff
@@ -158,6 +161,22 @@ declare_rustdoc_lint! {
     "detects URLs that could be written using only angle brackets"
 }
 
+declare_rustdoc_lint! {
+    /// The `doc_include` lint detects when `#[doc(include = ...)]` is used.
+    /// This feature is scheduled for removal and will give a hard error in a future release.
+    ///
+    /// This is a `rustdoc` only lint, see the documentation in the [rustdoc book].
+    ///
+    /// [rustdoc book]: ../../../rustdoc/lints.html#non_autolinks
+    DOC_INCLUDE,
+    Warn,
+    "detects using `#[doc(include = ...)]`",
+    @future_incompatible = FutureIncompatibleInfo {
+        reference: "issue #44732 <https://github.com/rust-lang/rust/issues/44732>",
+        edition: None,
+    };
+}
```

Comment:
This is such an overkill.
`doc(include)` is an unstable feature, it can be simply removed at any moment and that's all.

If you need to guide people to `#[doc = include_str!(...)]`, then message in `rustc_feature/src/removed.rs` gives opportunity to do that.

Reply:
>  doc(include) is an unstable feature, it can be simply removed at any moment and that's all.

This will break half of docs.rs, because it uses nightly and a lot of people don't test their documentation on nightly before publishing. If you have a suggestion other than a new lint I'm happy to hear it but I really don't think a hard error is the way to go.

Reply:
Alternatively, those users that don't test their documentation on nightly are unlikely to see this warning; while they are likely to notice that their documentation fails to build on docs.rs and fix it.

Reply:
I grepped for `doc(include` in all the crates on crates.io (with [`criner`](https://github.com/the-lean-crate/criner) and `find criner.db/assets | grep '\.crate$' | xargs zgrep -a 'doc(include' > doc-include.txt`). I came up with 6702 hits among 2618 crates; there are a total of 354798 crates.io releases, so that's almost a full percent of all crates that exist. I am not personally comfortable breaking that many crates.

For comparison, 348 of the latest versions use `doc(include)` and there are 58871 latest versions overall.

Reply:
Maybe a compromise would be to add a more general `rustdoc::deprecated_features` lint or something like that instead? Then we could use it for other features as well.

Reply:
@jyn514 
When do you plan to remove `doc(include)` from the compiler?
Unless you suggest to wait for years, only ~10-20 from these crates will notice the warning before the removal, we've seen similar numbers from the previous deprecation warnings, most of crates are not actively maintained and do not use `deny(warnings)` and can live with warnings for years.

I still suggest simply removing `doc(include)`, more people will notice that (maybe ~20-40 crates are somewhat maintained and will update).

(FWIW, the numbers from criner look too large, it would be good to manually verify what those 348 uses are.)

Reply:
I agree with @petrochenkov here (the important point here being that `extended_key_value_attributes` will be stabilized).